### PR TITLE
Created sub-total and total revenue for merchant dashboard

### PR DESCRIPTION
**Pull request by emaust**

### CODE REVIEW

**Review by tofuandeve — APPROVED** [review]
Approved!

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -69,6 +69,28 @@ def destroy
     return
   end
   
+  def buyer_info
+    @user_name = @order.buyer_name
+    @email = @order.buyer_email
+    @mailing_address = @order.buyer_address
+    @four_digits = @order.buyer_card.slice(-1..-4)
+    @cc_exp = @order.card_expiration
```
Comment (mdove92) [inline app/controllers/orders_controller.rb:77]: Impressive! Love the efficiency here!

File: app/views/merchants/show.html.erb [hunk 2]
```diff
@@ -58,12 +65,16 @@
     <% @paid_orders.each do |order| %>
       <p class="card-title">Order Number:  <%= order.id %></p>
       <% order.order_items.each do |item| %>
+        <%if item.product.merchant == @merchant%>
         <li class="card-text order-item">Name:  <%= Product.find_by(id: item.product_id).name %></li>
         <li class="card-text order-item">Quantity:  <%= "#{item.quantity}" %></li>
-        <li class="card-text order-item">Subtotal:  </li>
+        <li class="card-text order-item">Subtotal:  <%= "#{item.item_subtotal}" %></li>
+      <%end%>
```
Comment (mdove92) [inline app/views/merchants/show.html.erb:85]: HECK YEAH NESTED!

File: config/routes.rb [hunk 3]
```diff
@@ -22,6 +22,9 @@
     resources :products, only: [:index]
   end
   
+  get "/orders/:id/confirmtion", to: "orders#confirmation", as: "confirmation"
+  get "/orders/:id/buyer_info", to: "orders#buyer_info", as: "buyer_info"
```
Comment (mdove92) [inline config/routes.rb:26]: Nice complicated logic. Very beautiful.

**Review by mdove92 — APPROVED** [review]
Great work team! Thank you!

**Review by michaela260 — APPROVED** [review]
yay!

**Pull request merged**